sql/src/main/java/io/crate/analyze/AnalyzedColumnDefinition.java

```diff
@@ -175,15 +176,25 @@ public void validate() {
                             ident.sqlFqn()
                     ));
         }
-        if (isPrimaryKey() && collectionType != null) {
-            throw new UnsupportedOperationException(
-                    String.format(Locale.ENGLISH, "Cannot use columns of type \"%s\" as primary key", collectionType));
+        if (isPrimaryKey()) {
+            ensureNoCollectionOrObjectType();
         }
         for (AnalyzedColumnDefinition child : children) {
             child.validate();
         }
     }
 
+    private void ensureNoCollectionOrObjectType() {
```

Comment:
would be nice to have something like ensureTypeCanBeUsedAsKey
do we have the same restrictions like group by? probably field caches have the same restrictions?